src/Task/TwigCsFixer.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GrumPHP\Task;
+
+use GrumPHP\Formatter\ProcessFormatterInterface;
+use GrumPHP\Runner\TaskResult;
+use GrumPHP\Runner\TaskResultInterface;
+use GrumPHP\Task\Config\ConfigOptionsResolver;
+use GrumPHP\Task\Context\ContextInterface;
+use GrumPHP\Task\Context\GitPreCommitContext;
+use GrumPHP\Task\Context\RunContext;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use GrumPHP\Fixer\Provider\FixableProcessResultProvider;
+use Symfony\Component\Process\Process;
+
+/**
+ * @extends AbstractExternalTask<ProcessFormatterInterface>
+ */
+class TwigCsFixer extends AbstractExternalTask
+{
+    public static function getConfigurableOptions(): ConfigOptionsResolver
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults([
+            'path' => '.',
+            'triggered_by' => ['twig'],
+        ]);
+
+        $resolver->addAllowedTypes('path', ['string']);
+
+        return ConfigOptionsResolver::fromOptionsResolver($resolver);
+    }
+
+    public function canRunInContext(ContextInterface $context): bool
+    {
+        return $context instanceof GitPreCommitContext || $context instanceof RunContext;
+    }
+
+    public function run(ContextInterface $context): TaskResultInterface
+    {
+        $config = $this->getConfig()->getOptions();
+
+        $files = $context->getFiles()->extensions($config['triggered_by']);
+        if (0 === \count($files)) {
+            return TaskResult::createSkipped($this, $context);
+        }
+
+        $arguments = $this->processBuilder->createArgumentsForCommand('twig-cs-fixer');
+        $arguments->add('lint');
+        $arguments->add('--report=text');
+
+        if ($context instanceof GitPreCommitContext) {
+            $arguments->addFiles($files);
+        }
+
+        if ($context instanceof RunContext) {
+            $arguments->add($config['path']);
```

Comment:
Is it correct to assume paths is only being used in run context?
One could expect the pats to work as some kinf of filter in pre-commit context as well.
For example, you configure it as:

```yaml
paths:
  - templates/only-check-these-once
```

When committing `templates/unchecked-folder/file.twig` you don't expect the fixer to run on this?


Other tasks have things like:

- path black / white-listst
- the php-cs-fixer task has something like path intersection mode on the provided finder
- ...

Reply:
Now I've used the array of paths to filter the files that will be considered in the precommit. For example, if the only valid path is 'templates/' and I modify the file src/file.twig, it will be discarded since it is not found in the valid path.

Does it make sense? 

https://github.com/phpro/grumphp/pull/1131/files#r1601843153